### PR TITLE
[KeyVault] support async API design of security domain

### PR DESCRIPTION
**Pull request by isra-fel**
## Description

KeyVault will update their API design of security domain import/export to be asynchronous.
This PR make sure our cmdlet won't break when the API change takes place.

Change list:
- Support both sync and async API design
- Improved some error handling to throw AzPS*Exception
- Supported cancelling cmdlet execution by ctrl-C
- Added debug log

Testing:
I tested on public cloud for the old API. Will ask KeyVault team's help to test in dogfood.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by isra-fel** [comment]
Sign pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=767604&view=logs&j=275f1d19-1bd8-5591-b06b-07d489ea915a&t=771d55ec-34a7-5ddd-398e-3e8e57b4f3f0

File: src/KeyVault/KeyVault/SecurityDomain/Models/SecurityDomainClient.cs [hunk 1]
```diff
@@ -59,67 +64,178 @@ public string DownloadSecurityDomain(string hsmName, IEnumerable<X509Certificate
...
-        }
-
-        private void ValidateDownloadSecurityDomainResponse(SecurityDomainWrapper securityDomainWrapper)
-        {
-            if (string.IsNullOrEmpty(securityDomainWrapper.value) || !ValidateSecurityDomainData(securityDomainWrapper.value))
+            catch (Exception ex)
             {
-                _writeDebug($"Invalid security domain response: {securityDomainWrapper.value}");
-                throw new Exception(Resources.DownloadSecurityDomainFail);
+                throw new AzPSException(Resources.InvalidSubscriptionState, ErrorKind.UserError, ex);
```
Comment (BethanyZhou) [inline src/KeyVault/KeyVault/SecurityDomain/Models/SecurityDomainClient.cs:117]: ErrorKind.UserError doesn't match all cases

Comment (isra-fel) [reply src/KeyVault/KeyVault/SecurityDomain/Models/SecurityDomainClient.cs:117]: Makes sense. I'll change it to internal error

**Review by BethanyZhou — APPROVED** [review]
LGTM

**Pull request merged**